### DISCUSSION TITLE
`deleteinvoice` method

### DISCUSSION THREAD

**Discussion by fiatjaf** [ACINQ/eclair#1948] [Ideas]
Custodial services that deal with user accounts need a way to delete invoices. For example, if account A creates an invoice and sends it to B, then B pays it using his account on the same custodial service, then the payment must be settled internally and not through the Lightning Network.

When that happens, the service must also "delete" or "invalidate" the invoice on Éclair such that no one else can pay that same invoice from the external Lightning world.

If there is another way to tackle this issue I would be interested in it too.

**Comment by t-bast** [comment]
There's already a pending issue for that if anyone wants to start implementing: https://github.com/ACINQ/eclair/issues/1510
